muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/evaluation/dialogs/delete.tsx

```diff
@@ -0,0 +1,228 @@
+import Dialog from "~/components/general/dialog";
+import { connect, Dispatch } from "react-redux";
+import { bindActionCreators } from "redux";
+import Button from "~/components/general/button";
+import { AnyActionType } from "~/actions";
+import { StateType } from "~/reducers";
+import "~/sass/elements/form-elements.scss";
+import "~/sass/elements/form.scss";
+import { EvaluationState } from "../../../reducers/main-function/evaluation/index";
+import {
+  RemoveWorkspaceEvent,
+  removeWorkspaceEventFromServer,
+} from "../../../actions/main-function/evaluation/evaluationActions";
+import { i18nType } from "../../../reducers/base/i18n";
+import { EvaluationEvent } from "~/@types/evaluation";
+
+/**
+ * DeleteDialogProps
+ */
+interface DeleteDialogProps {
+  i18n: i18nType;
+  children: React.ReactElement<any>;
+  isOpen?: boolean;
+  onClose?: () => any;
+  evaluations: EvaluationState;
+  eventData: EvaluationEvent;
+  removeWorkspaceEventFromServer: RemoveWorkspaceEvent;
+}
+
+/**
+ * DeleteDialogState
+ */
+interface DeleteDialogState {}
+
+/**
+ * DeleteDialog
+ */
+class DeleteDialog extends React.Component<
+  DeleteDialogProps,
+  DeleteDialogState
+> {
+  /**
+   * constructor
+   * @param props
+   */
+  constructor(props: DeleteDialogProps) {
+    super(props);
+
+    this.handleDeleteEventClick = this.handleDeleteEventClick.bind(this);
+  }
+
+  /**
+   * handleDeleteEventClick
+   */
+  handleDeleteEventClick(closeDialog: () => any) {
+    const { eventData } = this.props;
+
+    this.props.removeWorkspaceEventFromServer({
+      identifier: eventData.identifier,
+      eventType: eventData.type,
+      onSuccess: () => {
+        const draftId = eventData.identifier;
+
+        cleanWorkspaceAndSupplementationDrafts(draftId);
+
+        closeDialog();
+      },
+      onFail: () => closeDialog(),
+    });
+  }
+
+  /**
+   * Component render method
+   * @returns JSX.Element
+   */
+  render() {
+    const { evaluationSelectedAssessmentId } = this.props.evaluations;
+
+    const studentNameString = `${evaluationSelectedAssessmentId.lastName}, ${evaluationSelectedAssessmentId.firstName}`;
+
+    let footer = (closeDialog: () => any) => {
+      return (
+        <div className="dialog__button-set">
+          <Button
+            buttonModifiers={["fatal", "standard-ok"]}
+            onClick={this.handleDeleteEventClick.bind(this, closeDialog)}
+          >
+            {this.props.i18n.text.get(
+              "plugin.evaluation.workspaceEvaluationDialog.removeDialog.removeButton",
+              studentNameString
+            )}
+          </Button>
+          <Button
+            buttonModifiers={["cancel", "standard-cancel"]}
+            onClick={closeDialog}
+          >
+            {this.props.i18n.text.get(
+              "plugin.evaluation.workspaceEvaluationDialog.removeDialog.cancelButton"
+            )}
+          </Button>
+        </div>
+      );
+    };
+    let content = (closeDialog: () => any) => {
```

Comment:
All these kind of let even in the body of render we should aim to change into const.

Reply:
Fixed